hastory-server/src/Data/Hastory/Server/Utils.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Utils where
+
+import Crypto.Hash (Digest, SHA256(..), hashWith)
+import qualified Data.ByteString as B
+import Data.Hastory.Types (Entry(..), ServerEntry(..), SyncRequest(..))
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+
+hashEntry :: Entry -> T.Text -> Digest SHA256
+hashEntry Entry {..} host = hashWith SHA256 (unifiedData :: B.ByteString)
+  where
+    unifiedData =
+      mconcat
+        [ T.encodeUtf8 entryText
+        , (hashPrepare . show) entryWorkingDir
+        , (hashPrepare . show) entryDateTime
```

Comment:
For this one you should use `formatTime` with some sensible time format so that it doesn't change if the `show` implementation changes.

Reply:
It looks like they made it much easier to convert a `UTCTime` value to the iso8601 format in [new version of the time package](https://www.stackage.org/haddock/lts-15.10/time-1.9.3/Data-Time-Format-ISO8601.html#v:iso8601Format).

Because we don't have that version available in our stackage resolver, I went with something more hand-rolled.